### PR TITLE
Unbreak minor versioning

### PR DESCRIPTION
**Pull request by maroux**
Since minor version changes ought to be backwards compatible, that implies a publisher must be able to publish new minor version without any changes to consumers. However, that's not currently possible since validation would fail on missing schema for the new minor version. This is fixed by ensuring that only major version is part of the schema file, and any changes are applied in-place.

### CODE REVIEW

File: docs/usage.rst [hunk 1]
```diff
@@ -45,5 +45,24 @@ A consumer for Lambda based workers can be started as following:
...
-.. _lambda sns format: https://docs.aws.amazon.com/lambda/latest/dg/eventsources.html#eventsources-sns
+Schema
+++++++
+
+The schema file must be a JSON-Schema `draft v4`_ schema. There's a few more restrictions in addition to being
+a valid schema:
+
+- There must be a top-level key called ``schemas``. The value must be an object.
+- ``schemas``: The keys of this object must be message types.  The value must be an object.
+- ``schemas/user_created``: The keys of this object must be major version types for this message type. The
```
Comment (mbachry) [inline docs/usage.rst]: Maybe a note above that `user_created` is just an example? It should be obvious for the reader, but mixing required name `schemas` with arbitrary name `user_created` may be confusing for some.

File: hedwig/validator.py [hunk 2]
```diff
@@ -14,6 +15,9 @@ class MessageValidator(Draft4Validator):
     # uuid separated by hyphens:
     _human_uuid_re = re.compile("^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$")
 
+    # schema parsing re, eg: hedwig.automatic.com/schema#/schemas/trip.created/1.0
+    _schema_re = re.compile(r'/([^/]+)$')
```
Comment (mbachry) [inline hedwig/validator.py]: `message.schema.split('/')[-1]` would be arguably simpler

**Comment by mbachry** [comment]
`x-versions` isn't used anywhere is code. Is it planned for another pull request in the future?

**Comment by maroux** [comment]
> x-versions isn't used anywhere is code. Is it planned for another pull request in the future?

This is intentional - `x-versions` is optional and only provided as guidance for book-keeping. I guess you could use it during `publish` to ensure that the full version is included in this list, but not sure if it's valuable.. thoughts?

File: hedwig/validator.py (hunk 2 above)
Comment (maroux) [reply hedwig/validator.py]: Yep, updated.

File: docs/usage.rst (hunk 1 above)
Comment (maroux) [reply docs/usage.rst]: Good call.

**Pull request merged**